### ISSUE TITLE
mergeData doesn't return expected results with age

### ISSUE THREAD

**Issue by jwokaty** [waldronlab/curatedMetagenomicData#227]
I noticed when using `mergeData` that I didn't get what I expected for `age`.

```{r}
> fm <- dplyr::filter(combined_metadata, age == 40, smoker == "yes", gender == "male")
> e <- filterByDataType(fm, "marker_presence") # reworked "selectWhere"
> filterByDataType
function(metadata, dataType) {
    filteredMetadata <- unique(unlist(metadata$studyName, use.names = FALSE)) %>%
        paste(dataType, sep = ".") %>%
        sapply(curatedMetagenomicData, dryrun = FALSE) %>%
        mergeData()
    filteredMetadata[ , metadata$sampleID]
}
> fm$age
[1] 40 40 40
> fm$subjectID
[1] "SAMEA104142325" "Travel006"      "wHAXPI043593-9"
> colData(e)[colData(e)$subjectID == "SAMEA104142325", ]$age
[1] NA
> colData(e)[colData(e)$subjectID == "wHAXPI043593-9", ]$age
[1] 40
```
Looking at [SAMEA104142325 in JieZ_2017](https://github.com/waldronlab/curatedMetagenomicDataCuration/blob/master/inst/curated/JieZ_2017/JieZ_2017_metadata.tsv#L351), I see that age has a decimal.

Out of curiosity, I tried

```{r}
> t <- curatedMetagenomicData("JieZ_2017.marker_presence", dryrun = FALSE) %>% mergeData()
```
and `age` isn't available as a column for `JieZ_2017.marker_presence` but it is for other studies like `AsnicarF_2020.marker_presence` when I run it through `mergeData`. Is there a reason for this (and it's expected) or is this a problem?

**Comment by lwaldron** [comment]
The cmd1 function gave correct results and had unit tests, and it's risky to replace it with this new untested function. This new function is now also in use to merge combined_metadata. @schifferl, the restoration of all the previous unit tests should be a really top priority. The previous merging functions could be used (without exporting) for additional unit tests.

**Comment by jwokaty** [comment]
I think this is the same issue @paolinomanghi documented at https://github.com/waldronlab/curatedMetagenomicDataCLI/issues/2.

**Comment by sdgamboa** [comment]
Closed by commit 68c96bb. See unit test [test-returnSamples](https://github.com/waldronlab/curatedMetagenomicData/blob/master/tests/testthat/test-returnSamples.R).
